### PR TITLE
Rebalance collateral work

### PR DESCRIPTION
**Pull request by UsamaSaleem14**
1. Rebalance Collateral work
2. Add Liquidity amount from frontend added while sending the transaction

### CODE REVIEW

**Comment by Pochetes** [comment]
Can you provide screenshots for this and elaborate more on how you got to your results?

File: src/hooks/useAddLiquidityHandler.tsx [hunk 1]
```diff
@@ -202,6 +214,7 @@ export const useAddLiquidityHandler = () => {
         console.log(args.reservesLen.toNumber())
         console.log(args.shares.toNumber())
       } catch (e) {
+        console.log(e)
```
Comment (TBomberman) [inline src/hooks/useAddLiquidityHandler.tsx:217]: Use notifyError.

File: src/hooks/useRebalanceHandler.tsx [hunk 2]
```diff
@@ -21,6 +29,10 @@ export const useRebalanceHandler = () => {
   const [slippage, setSlippage] = useState<string>('')
   const [slippageMinutes, setSlippageMinutes] = useState<string>('')
 
+  const [posManager, setPosManager] = useState<Contract | null>(null)
```
Comment (TBomberman) [inline src/hooks/useRebalanceHandler.tsx:32]: you can default this value to line 12

File: src/hooks/useRebalanceHandler.tsx [hunk 3]
```diff
@@ -21,6 +29,10 @@ export const useRebalanceHandler = () => {
...
+  const [gammaPool, setGammaPool] = useState<Contract | null>(null)
```
Comment (TBomberman) [inline src/hooks/useRebalanceHandler.tsx:33]: why do we need gamma pool now? should use everything from posmgr.

File: src/components/AddLiquiditySubmitButton.tsx [hunk 4]
```diff
@@ -1,15 +1,15 @@
...
 import { useAddLiquidityHandler } from '../hooks/useAddLiquidityHandler'
 
 type ButtonSubmitProps = {
-  canClick: Boolean
+  clickFunction: MouseEventHandler<HTMLDivElement>
   buttonStyle: string
   buttonText: string
 }
 
-const AddLiquiditySubmitValidator = ({ canClick, buttonStyle, buttonText }: ButtonSubmitProps): JSX.Element => {
```
[the remark this answers is not in the document]
Comment (UsamaSaleem14) [reply src/components/AddLiquiditySubmitButton.tsx:9]: Yeah we can remove this file and add this on the same page

File: src/hooks/useRebalanceHandler.tsx [hunk 5]
```diff
@@ -1,9 +1,17 @@
-import { ChangeEvent, Dispatch, SetStateAction, useCallback, useContext, useState } from 'react'
+import { ChangeEvent, Dispatch, SetStateAction, useCallback, useContext, useEffect, useState } from 'react'
 import Tokens, { Token } from '../components/Tokens'
 import { WalletContext } from '../context/WalletContext'
+import PositionManager from '../../abis/v1-periphery/PositionManager.sol/PositionManager.json'
+import GammaPool from '../../abis/v1-core/GammaPool.sol/GammaPool.json'
+import { Contract, ethers } from 'ethers'
+import toast from 'react-hot-toast'
```
[the remark this answers is not in the document]
Comment (UsamaSaleem14) [reply src/hooks/useRebalanceHandler.tsx:7]: Ok I will change it to use notification

File: src/hooks/useAddLiquidityHandler.tsx (hunk 1 above)
Comment (UsamaSaleem14) [reply src/hooks/useAddLiquidityHandler.tsx:217]: Ok

File: src/hooks/useRebalanceHandler.tsx (hunk 3 above)
Comment (UsamaSaleem14) [reply src/hooks/useRebalanceHandler.tsx:33]: In rebalance I need cfmm address. You said that I can get it from .cfmm() in GammaPool that's why I am using it

Comment (TBomberman) [reply src/hooks/useRebalanceHandler.tsx:33]: ok

File: src/hooks/useRebalanceHandler.tsx (hunk 2 above)
Comment (UsamaSaleem14) [reply src/hooks/useRebalanceHandler.tsx:32]: This is of contract type so we can not add string to it.

File: src/hooks/useAddLiquidityHandler.tsx [hunk 6]
```diff
@@ -72,23 +72,35 @@ export const useAddLiquidityHandler = () => {
...
+          buttonStyle={style.invalidatedButton}
+          buttonText={'Select Token'}
+        />
       )
     }
     if (tokenAInputVal === '' || tokenBInputVal === '') {
       return (
         <AddLiquiditySubmitButton
-          canClick={false}
+          clickFunction={doNothing}
```
[the remark this answers is not in the document]
Comment (UsamaSaleem14) [reply src/hooks/useAddLiquidityHandler.tsx]: I have removed this. We do not need this as we are not using this button. We have added this code in the same file of add liquidity

File: src/hooks/useAddLiquidityHandler.tsx (hunk 1 above)
Comment (UsamaSaleem14) [reply src/hooks/useAddLiquidityHandler.tsx]: Done

File: src/hooks/useRebalanceHandler.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (UsamaSaleem14) [reply src/hooks/useRebalanceHandler.tsx]: Done

File: src/components/AddLiquiditySubmitButton.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (UsamaSaleem14) [reply src/components/AddLiquiditySubmitButton.tsx:9]: Removed this file and wrote the code in the same page

[the remark this answers is not in the document]
Comment (TBomberman) [reply src/components/AddLiquiditySubmitButton.tsx:9]: I still see the Button in the components folder.

**Comment by TBomberman** [comment]
In the screenshots I see some colors are different from our normal theming. Can you change them to the same colors as the other pages?

File: src/components/AddLiquiditySubmitButton.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (UsamaSaleem14) [reply src/components/AddLiquiditySubmitButton.tsx:9]: Removed the file

**Comment by TBomberman** [comment]
image
This is the same color as figma, but not the same as add liquidity. Ok to commit @Pochetes ?

**Comment by Pochetes** [comment]
The theming is not final, we need to change the color scheme a bit. We should worry about functionality for this sprint.

**Review by Pochetes — COMMENTED** [review]
there were many errors as soon as I clicked on `rebalance`. Need to check that out. 

Also, it's important to guide us to the changes you made and explain code changes. It's easy to get lost as a reviewer and causes us to take longer to give a good review.

**Comment by TBomberman** [comment]
So far we are just connecting to the contract. We haven't integration tested it yet. Should do after the solidity unit tests are done to ensure they work on the tests at least.

**Comment by dam417** [comment]
Approving this PR so we can move to next step. Once smart contract is unit tested we can resolve any issues we find here since we'll know issues are here. The standard for now is as long as it's not crashing the front end it's fine.

**Review by dam417 — APPROVED** [review]
Approving this PR so we can move to next step. Once smart contract is unit tested we can resolve any issues we find here since we'll know issues are here. The standard for now is as long as it's not crashing the front end it's fine.

**Pull request merged**